### PR TITLE
Add json support for setup/stop/delete/cleanup commands

### PR DESCRIPTION
**Pull request by guillaumerose**
**Relates to:** Issue #1227

* Sometimes we prompt the user a yes/no question. With json output, he has to add ` --force` (a little bit like mv/cp).

### CODE REVIEW

File: pkg/crc/input/input.go [hunk 1]
```diff
@@ -1,17 +1,23 @@
...
 	survey "gopkg.in/AlecAivazis/survey.v1"
 )
 
 func PromptUserForYesOrNo(message string, force bool) bool {
 	if force {
 		return true
 	}
+	if !terminal.IsTerminal(int(os.Stdin.Fd())) {
+		return false
+	}
```
Comment (cfergeau) [inline pkg/crc/input/input.go:20]: Do we have the equivalent of `yum --assumeyes`/`yum --assumeno`?

File: cmd/crc/cmd/delete_test.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+	require.NoError(t, err)
+	defer os.RemoveAll(cacheDir)
+
+	out := new(bytes.Buffer)
+	assert.NoError(t, runDelete(out, fakemachine.NewClient(), true, cacheDir, false, true, jsonFormat))
+	assert.Equal(t, `{
+  "success": true
+}
+`, out.String())
+
```
Comment (cfergeau) [inline cmd/crc/cmd/delete_test.go:38]: Did you try `assert.JSONEq(t, ``{"success": true}``, out.String())` to avoid the slightly hard to parse/read multiline string?

**Review by cfergeau — COMMENTED** [review]
A few comments, still thinking that shortlog-only commits should be the exception rather than the norm :)

File: cmd/crc/cmd/setup.go [hunk 3]
```diff
@@ -31,13 +36,30 @@ func runSetup(arguments []string) error {
...
-		return err
+	err := preflight.SetupHost()
+	return render(&setupResult{
+		Success: err == nil,
+		Error:   errorMessage(err),
+	}, os.Stdout, outputFormat)
+}
+
+type setupResult struct {
+	Success bool   `json:"success"`
```
[the remark this answers is not in the document]
Comment (guillaumerose) [reply cmd/crc/cmd/setup.go:47]: sucess/errors was already present in the daemon API. I just kept it. If there is no error, the JSON output would be `{}` or `{errors:""}` ? It feels empty :)

File: pkg/crc/input/input.go (hunk 1 above)
Comment (guillaumerose) [reply pkg/crc/input/input.go:20]: No we don't have this.

File: cmd/crc/cmd/delete_test.go (hunk 2 above)
Comment (guillaumerose) [reply cmd/crc/cmd/delete_test.go:38]: I didn't know this assertions, thanks! I will use it now.

**Review by cfergeau — APPROVED** [review]
Looks good to me
>  Sometimes we prompt the user a yes/no question. With json output, he has to add --force (a little bit like mv/cp).

This could be described in the commit log for the stop command and the one changing getPullSecret/yes-no rather than just in the PR description
